docs/fuzzing.md

```diff
@@ -25,53 +25,77 @@ branch-tuple counts produces a control-flow "signature" for a run, which is used
 to differentiate runs / explore the control-flow space.
 
 Every run of the program starts anew and is fed a small binary input from the
-corpus. So to make this all work the program has to (a) run quickly and (b) take
-small inputs in binary form. We've modified stellar-core to have a mode for (b)
-but (a) is still unsatisfactory; the binary-input mode starts up a pair of
-Application objects in loopback configuration, and reads-and-receives a
-StellarMessage into them. This is quite a lot of setup and the fuzzing speed of
-AFL suffers as a result (it actually complains that the target is too slow).
+corpus. This initial corpus is the result of minimizing (using [afl-cmin][3])
+a much larger, randomly generated series of inputs, taking the smallest subset
+that results in unique control-flow tuples.
+
+To make this all work the program has to (a) run quickly, (b) take small inputs
+in binary form and (c) be guided toward interesting edge cases just enough. We
+currently have two fuzz modes, `tx` and `overlay`, and they **do not** perform the
+above equally well. For starters, we have enabled [llvm_mode][9]'s persistent mode
+for a modest 10x-100x improvement in execution/sec for *both* fuzz modes. Thus both
+perform well in area (a) and with simple modifications we made to stellar-core, (b)
+too. However, we've spent more energy with (c) for the `tx` fuzz mode, modifying
+stellar-core such that it is more deterministic -- caches are cleared and randomness
+seeded -- and it skips wasteful and inconsequential processes -- anything related to
+signatures. For both modes, there is still much room for improvement in regards to
+(a) and (c), one example being an isolation of the subsystem.
 
 
 ## Installing AFL
 
-Go to the [AFL website][0], download the [tarball][2], unpack and run `make`,
-then `sudo make install`. This will install both `afl-fuzz`, the fuzzer itself,
-and `afl-gcc` and `afl-clang`, which are compiler-wrappers that instrument
-binaries they compile with the fuzzer's branch-tracking machinery.
+Go to the [AFL repo][0], download the [tarball][2], unpack and run `make`, `make -C llvm_mode`
+then `sudo make install`. This will install both `afl-fuzz`, the fuzzer itself, and `afl-gcc`,
+`afl-clang`, and `afl-clang-fast`, which are compiler-wrappers that instrument binaries they
+compile with the fuzzer's branch-tracking machinery (the later being necessary for `llvm_mode`
+improvements described above).
 
 
 ## Building an instrumented stellar-core
 
-Start with a clean workspace, `make clean` or cleaner; then just do `./configure
---enable-afl && make` and make sure you have not enabled `asan` and `ccache`;
-the former is incompatible, and the latter doesn't interoperate with the
-compiler wrappers.
+Start with a clean workspace, `make clean` or cleaner; enable a few compilation flags **WHICH SHOULD
+NEVER BE ENABLED IN PRODUCTION**: `FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION` and `AFL_LLVM_MODE`; and
```

Comment:
are you sure about this? `FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION` seems to be set automatically by afl.

As for `AFL_LLVM_MODE`, if it's not automatically set, we should just set it when configuring with `--enable-afl` in the configure script